plugins/inputs/ipmi_sensor/ipmi.go

```diff
@@ -143,13 +159,76 @@ func (m *Ipmi) parse(acc telegraf.Accumulator, server string) error {
 	return nil
 }
 
-func Atofloat(val string) float64 {
+func parseV2(acc telegraf.Accumulator, hostname string, cmdOut string) error {
+	// each line will look something like
+	// CMOS Battery     | 65h | ok  |  7.1 |
+	// Temp             | 0Eh | ok  |  3.1 | 55 degrees C
+	// Drive 0          | A0h | ok  |  7.1 | Drive Present
+	lines := strings.Split(cmdOut, "\n")
+	for i := 0; i < len(lines); i++ {
+		ipmiFields := extractFieldsFromRegex(`^(?P<name>[^|]*)\|[^|]+\|(?P<status_code>[^|]*)\|(?P<entity_id>[^|]*)\|(?:(?P<description>[^|]+))?`, lines[i])
+		if len(ipmiFields) < 3 || len(ipmiFields) > 4 {
+			continue
+		}
+
+		tags := map[string]string{
+			"name": transform(ipmiFields["name"]),
+		}
+
+		// tag the server is we have one
+		if hostname != "" {
+			tags["server"] = hostname
+		}
+		tags["entity_id"] = transform(ipmiFields["entity_id"])
+		tags["status_code"] = trim(ipmiFields["status_code"])
+		fields := make(map[string]interface{})
+		descriptionResults := extractFieldsFromRegex(`^(?P<analogValue>[0-9.]+)\s(?P<analogUnit>.*)|(?P<status>.+)|^$`, trim(ipmiFields["description"]))
+		// This is an analog value with a unit
+		if descriptionResults["analogValue"] != "" && len(descriptionResults["analogUnit"]) >= 1 {
+			fields["value"] = aToFloat(descriptionResults["analogValue"])
+			// Some implementations add an extra status to their analog units
+			unitResults := extractFieldsFromRegex(`^(?P<realAnalogUnit>[^,]+)(?:,\s*(?P<statusDesc>.*))?`, descriptionResults["analogUnit"])
+			tags["unit"] = transform(unitResults["realAnalogUnit"])
+			if unitResults["statusDesc"] != "" {
+				tags["status_desc"] = transform(unitResults["statusDesc"])
+			}
+		} else {
+			// This is a status value
+			fields["value"] = 0.0
+			// Extended status descriptions aren't required, in which case for consistency re-use the status code
+			if descriptionResults["status"] != "" {
+				tags["status_desc"] = transform(descriptionResults["status"])
+			} else {
+				tags["status_desc"] = transform(ipmiFields["status_code"])
+			}
+		}
+
+		acc.AddFields("ipmi_sensor", fields, tags, time.Now())
+	}
+
+	return nil
+}
+
+// extractFieldsFromRegex consumes a regex with named capture groups and returns a kvp map of strings with the results
+func extractFieldsFromRegex(regex string, input string) map[string]string {
+	re := regexp.MustCompile(regex)
```

Comment:
We should compile all the regex once at the package level, then reuse the compiled regex here.